syntaxhighlighter.php

```diff
@@ -681,6 +681,11 @@ function shortcode_hack( $content, $callback, $ignore_html = true ) {
 
 			// Normal, safe parsing
 			$content = do_shortcode( $content, true );
+
+			// Escape all shortcodes still pending so WP doesn't parse them for us
+			$orig_tagnames = array_keys( $orig_shortcode_tags );
+			$orig_regex    = '/' . get_shortcode_regex( $orig_tagnames ). '/';
+			$content       = preg_replace_callback($orig_regex, array( $this, 'shortcode_hack_escape_shortcode' ), $content);
```

Comment:
I tried using a simpler regex here, but.. I couldn't make it work in every case I tested. I accept suggestions, though. :) (especially to remove the need for "shortcode_hack_escape_shortcode" method)